apps/server/src/modules/tool/context-external-tool/service/tool-reference.service.ts

```diff
@@ -47,4 +56,17 @@ export class ToolReferenceService {
 
 		return toolReference;
 	}
+
+	private async determineToolConfigurationStatusThroughValidation(
```

Comment:
Can you move this to a new service called commonVersionService and check the the feature flag there. 
Can you rename ExternalToolVersionService to ExternalToolVersionIncrementService